File: 卡码网/C++/test10.cpp
//10. 运营商活动
#include <iostream>
using namespace std;

int main(){
	int m,k;
	while(scanf("%d %d\n",&m,&k)!=EOF){
		if(m==0&&k==0) break;
		int total=m;
		while(m>=k){
			int left=m/k;
			total+=left;
			m=left+m%k;
		}
		printf("%d\n",total);
	}
	return 0;
}

File: 卡码/KamaCoder/Go/test10.go
//10. 运营商活动
package main

import (
	"fmt"
)

func main(){
	for {
		var m,k int
		_,err:=fmt.Scan(&m,&k)
		if err!=nil||m==0&&k==0 {
			return
		}
		var total=m
		for m>=k {
			left:=m/k
			total+=left
			m=left+m%k
		}
		fmt.Println(total)
	}
}
